bee-message/src/message.rs

```diff
@@ -197,7 +208,11 @@ impl Packable for Message {
             nonce,
         };
 
-        let message_len = unpacker.counter();
+        let message_len = if let (Some(start), Some(end)) = (start_opt, unpacker.read_bytes()) {
```

Comment:
I would put this declaration inside the `VERIFY`.  Could avoid a `packed_len` in some cases